### PR TITLE
Add cozy mobile build :tada:

### PR DESCRIPTION
**Pull request by kosssi**
- [x] Add specific .gitignore
- [x] Add new page only on mobile
- [x] Update build
- [x] Add documentation to mobile part
- [x] Add Android build on Travis
- [x] Mutualise mobile main.jsx and front main.jsx

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,7 +79,7 @@
   "dependencies": {
     "babel-preset-es2015": "^6.18.0",
     "classnames": "^2.2.0",
-    "cozy-client-js": "^0.0.1",
+    "cozy-client-js": "https://github.com/cozy/cozy-client-js.git#99b935bff6b3c3b83c58ea0f975b90a13a44c570",
```
Comment (enguerran) [inline package.json]: I don't like this. We need to publish the correct dependency on npm and maybe add a note in the `README.md` to explain a workaround if needed.

File: yarn.lock [hunk 2]
```diff
@@ -1336,9 +1336,9 @@ cosmiconfig@^2.1.0, cosmiconfig@^2.1.1:
     parse-json "^2.2.0"
     require-from-string "^1.1.0"
 
-cozy-client-js@^0.0.1:
+"cozy-client-js@https://github.com/cozy/cozy-client-js.git#99b935bff6b3c3b83c58ea0f975b90a13a44c570":
```
Comment (enguerran) [inline yarn.lock]: cf comment in `package.json`

File: config/webpack.base.config.js [hunk 3]
```diff
@@ -6,11 +6,14 @@ const HtmlWebpackPlugin = require('html-webpack-plugin')
...
+const mobile = process.env.MOBILE || false
+const outputFolder = mobile ? 'mobile/www' : 'build'
+const entryFile = mobile ? '../mobile/src/main' : '../src/main'
 
 module.exports = {
-  entry: path.resolve(__dirname, '../src/main'),
+  entry: path.resolve(__dirname, entryFile),
   output: {
-    path: path.resolve(__dirname, '../build'),
+    path: path.resolve(__dirname, '../' + outputFolder),
```
Comment (m4dz) [inline config/webpack.base.config.js]: Better if written `path.resolve(__dirname, '..', outputFolder)`, nope?

File: config/webpack.hash.config.js [hunk 4]
```diff
@@ -2,14 +2,16 @@
...
+const outputFolder = mobile ? 'mobile/www' : 'build'
 
 module.exports = {
   plugins: [
     // Extracts Hash in external file for reference
     function () {
       this.plugin('done', (stats) => {
         fs.writeFileSync(
-          path.join(__dirname, '../build/assets.json'),
+          path.join(__dirname, '../' + outputFolder + '/assets.json'),
```
Comment (m4dz) [inline config/webpack.hash.config.js]: Like in _base_ config, but I ask myself if we shouldn't export those values in a dedicated file, to centralize them. Any preference?

File: config/webpack.base.config.js (hunk 3 above)
Comment (kosssi) [inline config/webpack.base.config.js]: done

File: config/webpack.hash.config.js (hunk 4 above)
Comment (kosssi) [inline config/webpack.hash.config.js]: done

**Pull request merged**